ppr-ui/src/composables/mhrRegistration/useNewMhrRegistration.ts

```diff
@@ -198,9 +200,44 @@ export const useNewMhrRegistration = () => {
   const fetchMhRegistrations = async (): Promise<void> => {
     const draftFilings = await getMhrDrafts()
     const myMhrHistory = await mhrRegistrationHistory()
-    setMhrTableHistory([...draftFilings, ...myMhrHistory])
+    const filteredMhrHistory = addHistoryDraftsToMhr(myMhrHistory, draftFilings)
+    setMhrTableHistory([...filteredMhrHistory])
   }
 
+  function addHistoryDraftsToMhr (mhrHistory: MhRegistrationSummaryIF[], mhrDrafts: MhrDraftTransferApiIF[]):
+    MhRegistrationSummaryIF[] {
+    const sorter = require('lodash')
+    const sortedDraftFilings = sorter.orderBy(mhrDrafts, ['createDateTime'], ['desc'])
+    // add drafts to Registrations..
+    const filteredMhrHistory = sorter
+      .orderBy(mhrHistory.filter(t => t.registrationDescription === 'REGISTER NEW UNIT'), ['createdDateTime'], ['desc'])
+    filteredMhrHistory.forEach(transfer => {
+      transfer.baseRegistrationNumber = transfer.mhrNumber
+      var mhrDrafts = sortedDraftFilings.filter(s => s.mhrNumber === transfer.mhrNumber)
+      if (mhrDrafts?.length > 0) {
+        transfer.hasDraft = true
+        transfer.changes = []
+        mhrDrafts.forEach(draft => {
+          const newDraft: MhRegistrationSummaryIF = {
+            mhrNumber: transfer.mhrNumber,
+            baseRegistrationNumber: transfer.mhrNumber,
+            submittingParty: draft.submittingParty,
+            clientReferenceId: transfer.clientReferenceId,
+            createDateTime: draft.createDateTime,
+            error: draft.error,
+            registrationDescription: draft.registrationDescription,
+            hasDraft: true,
+            ownerNames: '',
+            path: draft.path,
```

Comment:
Would `path` for the draft serve the same purpose as the `hasDraft` flag?
What does the api return in the event when there is no draft?

Reply:
Very Good Question... 

Q1: I am not sure to be honest... I am using hasDraft to ensure that I don't break any existing logic... 

Q2: The api has two options... If you include "collapse=true&pageNumber=1&fromUI=true"  in querystring than the "changes" element is included which contains history, but not drafts.  That is why we add drafts to the original query.. 

The Registrations table uses the changes element to determine if the item is collapsible.

Reply:
Ok sounds fine, we can give it a try 👍